src/PaymentProcessing/SpherePayment.php

```diff
@@ -0,0 +1,257 @@
+
+        if ($this->front == 'patient') {
+            // Add specific parameters for patient front to the $mainUrl
+            $this->mainUrl = $this->mainUrl . '&hide_ticket=y&cvc_help=y&avs=y&postal=y&fulladdress=y&show_email=y';
+        } else {
+            // Add specific parameters for clinic front to the $mainUrl
+            $operatorEntry = (new UserService())->getCurrentlyLoggedInUser();
+            $operatorName = $operatorEntry['fname'] . " " . $operatorEntry['lname'];
+            $this->mainUrl = $this->mainUrl . '&hide_ticket=y&hide_trxoperator=y&trxoperator=' . urlencode($operatorName);
+        }
+    }
+
+    /**
+     * Return sphere code in javascript block.
+     *
+     * @return string
+     */
+    public function renderSphereJs(): string
+    {
+        if ($this->front == 'patient') {
+            return '<script>' . $this->renderSphereJsPatientFront() . $this->renderSphereJsCore() . '</script>';
+        } else { // $this->front == 'clinic'
+            return '<script>' . $this->renderSphereJsClinicFront() . $this->renderSphereJsCore() . '</script>';
+        }
+    }
+
+    /**
+     * Return patient front specific sphere code in javascript block.
+     *
+     * @return string
+     */
+    private function renderSphereJsPatientFront(): string
+    {
+        return "
+            function sphereSuccess(encData) {
+                let oForm = document.forms['payment-form'];
+                oForm.elements['mode'].value = 'Sphere';
+
+                let inv_values = JSON.stringify(getFormObj('invoiceForm'));
+                document.getElementById('invValues').value = inv_values;
+
+                let hiddenInput = document.createElement('input');
+                hiddenInput.setAttribute('type', 'hidden');
+                hiddenInput.setAttribute('name', 'enc_data');
+                hiddenInput.setAttribute('value', encData);
+                oForm.appendChild(hiddenInput);
+
+                // Submit payment to server
+                fetch('./lib/paylib.php', {
+                    method: 'POST',
+                    body: new FormData(oForm)
+                }).then(function(response) {
+                    if (!response.ok) {
+                        throw Error(response.statusText);
+                    }
+                    return response.text();
+                }).then(function(data) {
+                    if(data !== 'ok') {
+                        alert(data);
+                        return;
+                    }
+                    alert(chargeMsg);
+                    window.location.reload(false);
+                }).catch(function(error) {
+                    alert(error)
```

Comment:
Do you not want to do a reload here? in the above then() clause if you hit an error you still reload the page.